### ISSUE TITLE
addSelectionToNextFindMatch shortcut (ctrl+d) does not work with select entire line (shift + v)

### ISSUE THREAD

**Issue by ralphie02** [VSCodeVim/Vim#4895]
**Describe the bug**
Note that I have the following settings:
```
...
  "vim.useCtrlKeys": false,
  "vim.handleKeys": {
    "<C-c>": false,
    "<C-d>": false,
    "<C-u>": false,
    "<C-v>": false,
  },
...
```
When an entire line of text is selected with `shift`+`v` and when I try to select the next match with `ctrl`+`d`, the text does not get selected

In contrast, this works fine if the text is first selected with a mouse and `ctrl`+`d` is pressed
OR
the text is selected (in visual mode) with `v` + `nav key` (`h`,`j`,`k`,`l`) and `ctrl`+`d` is pressed

**To Reproduce**
1. Open/create a file that has lines of duplicated text.
ie.
```
sample text
sample text
sample text
....
```
2.  Navigate to the first line and press `shift`+`v`
3.  Once the line is selected as per step 2, press `ctrl`+`d`
4.  Watch as the shortcut fails

**Expected behavior**
I expect subsequent lines of duplicated text to be selected

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: 1.14.5 
- Version: 1.45.1 (user setup)
- OS: Windows_NT x64 10.0.18363

**Additional context**
Not sure if useful but I got this in my settings:

Also, this might be related/duplicate of issue #4277

**Comment by AkkadCultist** [comment]
Anyone working on this ?

**Comment by J-Fields** [comment]
Not that I'm aware of. Happy to accept PRs

**Comment by sql-koala** [comment]
Hi, I did look at the code. It cant work, because cmdD /gb is not enabled for visualLine mode. The reason may be, that vim gets weird, when you go to visualLine mode with multicusor. But the other way is not a problem, because cmdD will set mode to visual.

Enabling it, I found no problem. It would function like this:

```
sample text
    sample text
sample text xx
    sample text xx
sample text

```
- Starting on first line, it would add cursors to all
- starting second line would add 4. line
- starting 3 line would add 4.

If that is ok, it is easy to enable. If one wanted that cursor are only added on exactly the same lines, it could be a bit more work.

**Comment by ralphie02** [comment]
I'm sorry but I don't think I understand what you mean by `cmdD /gb`. Is this Mac's version of `ctrl` + `d`? Also, is gb referring to Great Britain? Is this specific to UK English Keyboard? I'm already at a loss base on the initial statement....

You mentioned that vim might be acting weird on `multicursor` mode, does this specific scenario _count_ as `multicursor`? I was under the impression that it's not `mutlicursor` so I don't think that should be related to this problem. If it IS in `multicursor` mode, the question I have is WHY? Does that even make sense? I'm coming from sublime with vim mode (ie. vintage) enabled and this functionality works fine there; I was hoping for the same behavior here.

I'm not sure what you mean by enabling it. I'm also not sure what your sample scenario describes. You didn't exactly explain what steps were taken.
> - Starting on first line, it would add cursors to all  

I'm guessing this means press `shift` + `v` ??
> - starting second line would add 4. line

lost...no idea what you mean. `Add 4. line`? What did you do to add 4 lines? What lines? Did you actually mean characters? 
> - starting 3 line would add 4.

same here, what got added?? 4 line??

Overall, I don't think I get what your proposed solution entails....Can you give me more info?

**Comment by J-Fields** [comment]
> I'm sorry but I don't think I understand what you mean by cmdD /gb. Is this Mac's version of ctrl + d? Also, is gb referring to Great Britain? Is this specific to UK English Keyboard? I'm already at a loss base on the initial statement....

`cmd-d` is `ctrl-d` on Mac. `gb` (press `g`, then press `b`) is a shortcut in VSCodeVim for the same action (`addSelectionToNextFindMatch`).

> You mentioned that vim might be acting weird on multicursor mode, does this specific scenario count as multicursor? I was under the impression that it's not mutlicursor so I don't think that should be related to this problem. If it IS in multicursor mode, the question I have is WHY? Does that even make sense?

Yeah this is multi-cursor, I'm not sure why it wouldn't be. `ctrl-d` adds a cursor each time you hit it.

> Overall, I don't think I get what your proposed solution entails....Can you give me more info?

He's saying that it would find instances of the contents of the original line, regardless of whether those instances are a complete line themselves. So with the cursor in visual line mode on the first line of the following text:

```
foobar
foobarXYZ
    ___foobar___
```

`ctrl-d` would put a cursor on the second line, and `ctrl-d again would put a cursor on the third.

**Comment by ralphie02** [comment]
@J-Fields You're right. For some reason I was stuck on the idea that selecting the entire line with `shift` + `v` is multicursor, not sure why I completely disregarded the part of my question talking about `ctrl` + `d`....

I guess based on the last statement, what I want is not easily done....

**Comment by J-Fields** [comment]
It's doable, but we would have to implement it from scratch rather than leveraging `addSelectionToNextFindMatch `

**Comment by ralphie02** [comment]
Went back to this issue and realized that you might have been waiting for my reply.

I'd still like for this to get implemented but wouldn't really care much if it gets completely ignored due to time and effort involved. I can always work around it.
